### ISSUE TITLE
Pulling images will result in errors sometimes, and sometimes not

### ISSUE THREAD

**Issue by gabrielbull** [cri-o/cri-o#8760]
### What happened?

Using the following command results in errors sometimes, all the time on large images:

```
crictl pull [image]
```

You will get this error: 

```
pulling image: rpc error: code = Canceled desc = copying system image from manifest list: copying config: context canceled
```

### What did you expect to happen?

crictl should pull the image, works fine all the time with contained.

### How can we reproduce it (as minimally and precisely as possible)?

Install cri-o and cri-tools on a fresh instance, tried both Amazon Linux 2023 and Ubuntu 22.04 with the same result. Pull an image using crictl.

### CRI-O and Kubernetes version

<details>

```console
crio version 1.31.2
   GitCommit:      920af25f2b6fd4f52bced0e73f696e48464da543
   GitCommitDate:  2024-11-01T10:41:59Z
   GitTreeState:   dirty
   BuildDate:      1970-01-01T00:00:00Z
   GoVersion:      go1.22.5
   Compiler:       gc
   Platform:       linux/amd64
   Linkmode:       static
   BuildTags:
     static
     netgo
     osusergo
     exclude_graphdriver_btrfs
     seccomp
     apparmor
     selinux
     exclude_graphdriver_devicemapper
   LDFlags:          unknown
   SeccompEnabled:   true
   AppArmorEnabled:  false
```

```console
{
  "clientVersion": {
    "major": "1",
    "minor": "31",
    "gitVersion": "v1.31.2",
    "gitCommit": "5864a4677267e6adeae276ad85882a8714d69d9d",
    "gitTreeState": "clean",
    "buildDate": "2024-10-22T20:35:25Z",
    "goVersion": "go1.22.8",
    "compiler": "gc",
    "platform": "linux/amd64"
  },
  "kustomizeVersion": "v5.4.2"
}
```

</details>

### OS version

<details>

```console
PRETTY_NAME="Ubuntu 24.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04.1 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo

Linux ip-172-31-38-233 6.8.0-1016-aws #17-Ubuntu SMP Mon Sep  2 13:48:07 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

```console
NAME="Amazon Linux"
VERSION="2023"
ID="amzn"
ID_LIKE="fedora"
VERSION_ID="2023"
PLATFORM_ID="platform:al2023"
PRETTY_NAME="Amazon Linux 2023.6.20241031"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2023"
HOME_URL="https://aws.amazon.com/linux/amazon-linux-2023/"
DOCUMENTATION_URL="https://docs.aws.amazon.com/linux/"
SUPPORT_URL="https://aws.amazon.com/premiumsupport/"
BUG_REPORT_URL="https://github.com/amazonlinux/amazon-linux-2023"
VENDOR_NAME="AWS"
VENDOR_URL="https://aws.amazon.com/"
SUPPORT_END="2028-03-15"

Linux ip-172-31-38-253.us-east-2.compute.internal 6.1.112-124.190.amzn2023.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Oct 23 06:32:04 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

</details>

### Additional environment details (AWS, VirtualBox, physical, etc.)

<details>

</details>

**Comment by saschagrunert** [comment]
@gabrielbull thank you for the issue! Do you have a reproducer for it?

**Comment by saschagrunert** [comment]
Assuming this is now fixed.

**Comment by f5-jay** [comment]
Do we know yet what targeted version(s) the fix will be in?

Will we see a fix in v1.29.x, v1.30.x, and v1.31.x or just v1.31.x?  They are all three impacted.

**Comment by f5-jay** [comment]
CRIO v1.32.0 has been released.  I can confirm the new flag is working in this release.

https://github.com/cri-o/cri-o/blob/main/docs/crio.conf.5.md

> pull_progress_timeout="10s" The timeout for an image pull to make progress until the pull operation gets canceled. This value will be also used for calculating the pull progress interval to pull_progress_timeout / 10. Can be set to 0 to disable the timeout as well as the progress output.


I added the following to /etc/crio/crio.conf

```
[crio.image]
pull_progress_timeout="0"
```
